grpc/src/main/java/io/stargate/grpc/codec/cql/CollectionCodec.java

```diff
@@ -0,0 +1,58 @@
+package io.stargate.grpc.codec.cql;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import io.stargate.db.schema.Column;
+import io.stargate.db.schema.Column.ColumnType;
+import io.stargate.proto.QueryOuterClass.Collection;
+import io.stargate.proto.QueryOuterClass.Value;
+import java.nio.ByteBuffer;
+
+public class CollectionCodec extends CompositeCodec {
+
+  @Override
+  public ByteBuffer encode(@NonNull Value value, @NonNull Column.ColumnType type) {
+    if (!value.hasCollection()) {
+      throw new IllegalArgumentException("Expected collection type");
+    }
+    assert type.isCollection();
```

Comment:
Why an assert rather than an actual check? A few lines down will throw an NPE if it's not a collection, right?

```java
Column.ColumnType elementType = type.parameters().get(0);
```

Reply:
+1 to converting `assert` to an always-on runtime check. It should not have any significant overhead in this case, right?

Reply:
This truly is an assert. The codec is determined by the type. Something's really gone wrong if that assert fails and it's really not possible.